src/components/ModalWithConfirm.js

```diff
@@ -0,0 +1,38 @@
+import Modal from "./Modal.js";
+
+class ModalWithConfirm extends Modal {
+  constructor({ modalSelector, handleConfirmAction }) {
+    //
+    super(modalSelector);
+    this._modalSelector = document.querySelector(modalSelector);
+    this._modalConfirmBox = this._modalSelector.querySelector(
+      ".modal__confirm-box"
+    );
+    this._handleConfirmAction = handleConfirmAction;
+    this.isSubmitted = false;
+  }
+  open(currentCard, cardId) {
+    this._currentCard = currentCard;
+    this._cardId = cardId;
+    super.open();
+  }
+
+  close() {
+    super.close();
+  }
+  getIsSubmitted() {
+    return this.isSubmitted;
+  }
+
+  setEventListeners() {
+    this._modalConfirmBox.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form